variables.tf

```
@@ -342,9 +342,9 @@ variable "terraform_smart_sanitization" {
 }
 
 variable "terraform_version" {
-  type        = string
-  description = "Terraform version to use."
-  default     = "1.7.2"
+  type        = optional(string)
```

The `optional` keyword is only usable in `object` types. Standard types in Terraform + OpenTofu allow `null`. See here: https://opentofu.org/docs/language/expressions/type-constraints/#optional-object-type-attributes

Let's change this back to `string` and your change should be all good 👍
```suggestion
  type        = string
```